src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -203,17 +203,18 @@ public Declaration.Scoped createHeader(Cursor c, List<Declaration> decls) {
 
     public Declaration.Scoped createRecord(Cursor c, Declaration.Scoped.Kind scopeKind) {
         Type.Declared t = (Type.Declared)RecordLayoutComputer.compute(typeMaker, 0, c.type(), c.type());
-        List<Declaration> decls = filterNestedDeclarations(t.tree().members());
         if (c.isDefinition()) {
+            Declaration.Scoped scoped = t.tree();
+            List<Declaration> decls = filterNestedDeclarations(scoped.members());
             //just a declaration AND definition, we have a layout
-            return Declaration.scoped(scopeKind, CursorPosition.of(c), c.spelling(),
-                                      t.tree().layout().get(), decls.toArray(new Declaration[0]));
+            return Declaration.scoped(scoped.kind(), scoped.pos(), scoped.name(),
+                                      scoped.layout().get(), decls.toArray(new Declaration[0]));
         } else {
             //if there's a real definition somewhere else, skip this redundant declaration
             if (!c.getDefinition().isInvalid()) {
                 return null;
             }
-            return Declaration.scoped(scopeKind, CursorPosition.of(c), c.spelling(), decls.toArray(new Declaration[0]));
+            return Declaration.scoped(scopeKind, CursorPosition.of(c), c.spelling());
```

Comment:
To be 100% fair, I don't understand which situation is the `return` in the `else` supposed to cover. Even the old code seems off. E.g. `RecordLayoutComputer` seems to filter out cases where `c.getDefinition().isInvalid()`. But for everything else there should be a layout. But the layout was thrown away in the old impl as well. So, my goal here was to have a cleaner separation between "supported" cases and "unsupported" ones. Note that if we create a scoped decl w/o a layout (as we do in the old code), that thing ends up being skipped anyway by `OutputFactory`:

```
if (d.layout().isEmpty() || structDefinitionSeen(d)) {
            //skip decl
            return null;
        }
 ```
 
Anyway, the main problem that was fixed here was to make sure that the _name_ of the type returned by `RecordLayoutComputer.compute` was reused, at least in the "supported" path. IMHO, we could just return `null` in the else, and we would probably not change semantics.

Reply:
Did some more analysis, which refreshed my memory. When the cursor is not a definition, record layout computer will just return a fake declared type, backed by a padding layout. It will **not** look at any of the sub-cursors (as there's no definition for clang to see), so there will be no members attached to the scoped tree of the declared type. In other words, `decls` is guaranteed to be empty here.

Reply:
> I thought perhaps the `else` was covering opaque types. i.e. in a case where we just have `typedef struct Opaque* opaque_t`, there would still be a `Declaration.Scoped` in the tree without a layout.

Yes and no - the main difference is that we tolerate opaque types when they are nested somewhere else. E.g. a pointer to an opaque type is a valid type for a struct field, or function parameter. Since the routine that create types ends up calling TreeMaker when it sees a struct/union, we have to do something for these types (other than returning `null`). But in such cases there' never a nested declaration attached to such types (because they are... opaque).